# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

if pkg-config --exists glib-build; then
  GLIB=1
  REQUIRES="$REQUIRES glib-build"
else
  GUDEV=--disable-gudev
fi

# configure (ac-2.69; am-1.13; lt-2.4.2)
libtoolize -c -f; aclocal; autoconf; autoheader -f; automake -a -c
./configure $GUDEV --disable-keymap --disable-static --with-noarch-pkgconfigdir=/lib/pkgconfig --sbindir=/bin --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/udev/mdevd $DESTDIR/bin
install src/udev/udevd $DESTDIR/bin
install src/udev/udevadm $DESTDIR/bin
ln -sf udevadm $DESTDIR/bin/mdev
install -d $DESTDIR/etc/udev
install -m 644 src/udev/udev.conf $DESTDIR/etc/udev
install -d $DESTDIR/etc/udev/rules.d
cat > $DESTDIR/etc/udev/rules.d/udev.rules << EOF
SUBSYSTEM=="input", IMPORT{program}="input_id %p"
EOF
install -d $DESTDIR/include
install -m 644 src/libudev/libudev.h $DESTDIR/include
test $GLIB && install -d $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudev.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevclient.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevdevice.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevenumerator.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevenums.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevenumtypes.h $DESTDIR/include/gudev-1.0/gudev
test $GLIB && install -m 644 src/gudev/gudevtypes.h $DESTDIR/include/gudev-1.0/gudev
install -d $DESTDIR/lib
test $GLIB && install src/gudev/.libs/libgudev-1.0.so.0.* $DESTDIR/lib/libgudev-1.0.so.0
test $GLIB && ln -sf libgudev-1.0.so.0 $DESTDIR/lib/libgudev-1.0.so
install src/libudev/.libs/libudev.so.0.* $DESTDIR/lib/libudev.so.0
ln -sf libudev.so.0 $DESTDIR/lib/libudev.so
install -d $DESTDIR/lib/udev
install src/ata_id/ata_id $DESTDIR/lib/udev
install src/cdrom_id/cdrom_id $DESTDIR/lib/udev
install src/collect/collect $DESTDIR/lib/udev
install src/scsi_id/scsi_id $DESTDIR/lib/udev
install src/v4l_id/v4l_id $DESTDIR/lib/udev
install -d $DESTDIR/lib/pkgconfig
test $GLIB && install -m 644 src/gudev/gudev-1.0.pc $DESTDIR/lib/pkgconfig
install -m 644 src/libudev/libudev.pc $DESTDIR/lib/pkgconfig
install -m 644 src/udev/udev.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/eudev-build.pc << EOF
Name: eudev
Version: 2.1.1
Description: Neutral Userspace Device management
Requires: $REQUIRES

devel=\\
/include/libudev.h \\
EOF
test $GLIB && echo /include/gudev-1.0/gudev/gudev.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevclient.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevdevice.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevenumerator.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevenums.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevenumtypes.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /include/gudev-1.0/gudev/gudevtypes.h \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
test $GLIB && echo /lib/libgudev-1.0.so \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
cat >> $DESTDIR/share/pkgconfig/eudev-build.pc << EOF
/lib/libudev.so \\
EOF
test $GLIB && echo /lib/pkgconfig/gudev-1.0.pc \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
cat >> $DESTDIR/share/pkgconfig/eudev-build.pc << EOF
/lib/pkgconfig/libudev.pc \\
/lib/pkgconfig/udev.pc

exec=\\
/bin/mdevd \\
/bin/udevd \\
/bin/udevadm \\
/etc/udev/udev.conf \\
/etc/udev/rules.d/udev.rules \\
EOF
test $GLIB && echo /lib/libgudev-1.0.so.0 \\ >> $DESTDIR/share/pkgconfig/eudev-build.pc
cat >> $DESTDIR/share/pkgconfig/eudev-build.pc << EOF
/lib/libudev.so.0 \\
/lib/udev/ata_id \\
/lib/udev/cdrom_id \\
/lib/udev/collect \\
/lib/udev/scsi_id \\
/lib/udev/v4l_id
EOF
test $GLIB && ln -sf eudev-build.pc $DESTDIR/share/pkgconfig/gudev-build.pc
ln -sf eudev-build.pc $DESTDIR/share/pkgconfig/libudev-build.pc
ln -sf eudev-build.pc $DESTDIR/share/pkgconfig/udevd-build.pc

exit
--- eudev-2.1.1.orig/configure.ac
+++ eudev-2.1.1/configure.ac
@@ -164,7 +164,7 @@
 
 # pkgconfigdir paths
 AC_SUBST([pkgconfiglibdir], [${libdir}/pkgconfig])
-AC_SUBST([sharepkgconfigdir],[${datadir}/pkgconfig])
+PKG_NOARCH_INSTALLDIR
 
 # gudev paths
 AC_SUBST([libgudev_includedir],[${includedir}/gudev-1.0/gudev])
--- eudev-2.1.1.orig/src/libudev/Makefile.am
+++ eudev-2.1.1/src/libudev/Makefile.am
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4 ${ACLOCAL_FLAGS}
 
-LIBUDEV_CURRENT=7
+LIBUDEV_CURRENT=6
 LIBUDEV_REVISION=0
 LIBUDEV_AGE=6
 
--- eudev-2.1.1.orig/src/libudev/libudev.sym
+++ eudev-2.1.1/src/libudev/libudev.sym
@@ -7,98 +7,10 @@
   (at your option) any later version.
 ***/
 
-LIBUDEV_183 {
-global:
-        udev_device_get_action;
-        udev_device_get_devlinks_list_entry;
-        udev_device_get_devnode;
-        udev_device_get_devnum;
-        udev_device_get_devpath;
-        udev_device_get_devtype;
-        udev_device_get_driver;
-        udev_device_get_is_initialized;
-        udev_device_get_parent;
-        udev_device_get_parent_with_subsystem_devtype;
-        udev_device_get_properties_list_entry;
-        udev_device_get_property_value;
-        udev_device_get_seqnum;
-        udev_device_get_subsystem;
-        udev_device_get_sysattr_list_entry;
-        udev_device_get_sysattr_value;
-        udev_device_get_sysname;
-        udev_device_get_sysnum;
-        udev_device_get_syspath;
-        udev_device_get_tags_list_entry;
-        udev_device_get_udev;
-        udev_device_get_usec_since_initialized;
-        udev_device_has_tag;
-        udev_device_new_from_devnum;
-        udev_device_new_from_environment;
-        udev_device_new_from_subsystem_sysname;
-        udev_device_new_from_syspath;
-        udev_device_ref;
-        udev_device_unref;
-        udev_enumerate_add_match_is_initialized;
-        udev_enumerate_add_match_parent;
-        udev_enumerate_add_match_property;
-        udev_enumerate_add_match_subsystem;
-        udev_enumerate_add_match_sysattr;
-        udev_enumerate_add_match_sysname;
-        udev_enumerate_add_match_tag;
-        udev_enumerate_add_nomatch_subsystem;
-        udev_enumerate_add_nomatch_sysattr;
-        udev_enumerate_add_syspath;
-        udev_enumerate_get_list_entry;
-        udev_enumerate_get_udev;
-        udev_enumerate_new;
-        udev_enumerate_ref;
-        udev_enumerate_scan_devices;
-        udev_enumerate_scan_subsystems;
-        udev_enumerate_unref;
-        udev_get_log_priority;
-        udev_get_userdata;
-        udev_list_entry_get_by_name;
-        udev_list_entry_get_name;
-        udev_list_entry_get_next;
-        udev_list_entry_get_value;
-        udev_monitor_enable_receiving;
-        udev_monitor_filter_add_match_subsystem_devtype;
-        udev_monitor_filter_add_match_tag;
-        udev_monitor_filter_remove;
-        udev_monitor_filter_update;
-        udev_monitor_get_fd;
-        udev_monitor_get_udev;
-        udev_monitor_new_from_netlink;
-        udev_monitor_receive_device;
-        udev_monitor_ref;
-        udev_monitor_set_receive_buffer_size;
-        udev_monitor_unref;
-        udev_new;
-        udev_queue_get_kernel_seqnum;
-        udev_queue_get_queue_is_empty;
-        udev_queue_get_queued_list_entry;
-        udev_queue_get_seqnum_is_finished;
-        udev_queue_get_seqnum_sequence_is_finished;
-        udev_queue_get_udev;
-        udev_queue_get_udev_is_active;
-        udev_queue_get_udev_seqnum;
-        udev_queue_new;
-        udev_queue_ref;
-        udev_queue_unref;
-        udev_ref;
-        udev_set_log_fn;
-        udev_set_log_priority;
-        udev_set_userdata;
-        udev_unref;
-        udev_util_encode_string;
-local:
-        *;
-};
-
 LIBUDEV_189 {
 global:
         udev_device_new_from_device_id;
-} LIBUDEV_183;
+};
 
 LIBUDEV_196 {
 global:
--- eudev-2.1.1.orig/src/mtd_probe/mtd_probe.h
+++ eudev-2.1.1/src/mtd_probe/mtd_probe.h
@@ -18,6 +18,7 @@
  */
 
 #include <mtd/mtd-user.h>
+#include <stdint.h>
 
 /* Full oob structure as written on the flash */
 struct sm_oob {
--- eudev-2.1.1.orig/src/udev/Makefile.am
+++ eudev-2.1.1/src/udev/Makefile.am
@@ -89,7 +89,7 @@
 dist_udevconf_DATA = \
 	udev.conf
 
-sharepkgconfig_DATA = \
+noarch_pkgconfig_DATA = \
 	udev.pc
 
 EXTRA_DIST = \
--- eudev-2.1.1.orig/src/udev/mdevd
+++ eudev-2.1.1/src/udev/mdevd
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+exec udevd -d
--- eudev-2.1.1.orig/src/udev/udevadm.c
+++ eudev-2.1.1/src/udev/udevadm.c
@@ -65,7 +65,7 @@
 static int adm_help(struct udev *udev, int argc, char *argv[]) {
         unsigned int i;
 
-        fprintf(stderr, "Usage: udevadm [--help] [--version] [--debug] COMMAND [COMMAND OPTIONS]\n");
+        fprintf(stderr, "Usage: udevadm [--help] [--version] [--debug] [--scan] COMMAND [COMMAND OPTIONS]\n");
         for (i = 0; i < ELEMENTSOF(udevadm_cmds); i++)
                 if (udevadm_cmds[i]->help != NULL)
                         printf("  %-12s %s\n", udevadm_cmds[i]->name, udevadm_cmds[i]->help);
@@ -85,6 +85,7 @@
         static const struct option options[] = {
                 { "debug", no_argument, NULL, 'd' },
                 { "help", no_argument, NULL, 'h' },
+                { "scan", no_argument, NULL, 's' },
                 { "version", no_argument, NULL, 'V' },
                 {}
         };
@@ -100,7 +101,7 @@
         udev_set_log_fn(udev, udev_main_log);
         mac_selinux_init("/dev");
 
-        while ((c = getopt_long(argc, argv, "+dhV", options, NULL)) >= 0)
+        while ((c = getopt_long(argc, argv, "+dhsV", options, NULL)) >= 0)
                 switch (c) {
 
                 case 'd':
@@ -112,6 +113,10 @@
                         rc = adm_help(udev, argc, argv);
                         goto out;
 
+                case 's':
+                        rc = run_command(udev, udevadm_cmds[1], argc, argv);
+                        goto out;
+
                 case 'V':
                         rc = adm_version(udev, argc, argv);
                         goto out;
